File: multisnaprecyclerview/src/main/java/com/takusemba/multisnaprecyclerview/SnapGravity.java
package com.takusemba.multisnaprecyclerview;

/**
 * SnapGravity
 *
 * @author takusemba
 * @since 30/07/2017
 **/
enum SnapGravity {

    /**
     * gravity to center
     */
    CENTER(0),

    /**
     * gravity to start (left or top)
     */
    START(1),

    /**
     * gravity to start (right or bottom)
     */
    END(2);

    private int value;

    SnapGravity(int value) {
        this.value = value;
    }

    public int getValue() {
        return this.value;
    }

    public static SnapGravity valueOf(int value) {
        for (SnapGravity gravity : SnapGravity.values()) {
            if (gravity.getValue() == value) {
                return gravity;
            }
        }
        throw new IllegalArgumentException("no such enum object for the value: " + value);
    }

}

File: multisnaprecyclerview/src/main/java/com/takusemba/multisnaprecyclerview/SnapGravity.kt
package com.takusemba.multisnaprecyclerview

/**
 * Supported gravity for snapping.
 */
enum class SnapGravity(val value: Int) {

  /**
   * gravity to center.
   */
  CENTER(0),

  /**
   * gravity to start (left or top).
   */
  START(1),

  /**
   * gravity to start (right or bottom).
   */
  END(2);

  companion object {

    fun valueOf(value: Int): SnapGravity {
      for (gravity in values()) {
        if (gravity.value == value) {
          return gravity
        }
      }
      throw IllegalArgumentException("no such enum object for the value: $value")
    }
  }
}
